server/base/src/main/java/org/apache/accumulo/server/conf/codec/PropSerdes.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.conf.codec;
+
+/**
+ * Serialize / Deserialize versioned properties into a byte array.
+ */
+public interface PropSerdes {
```

Comment:
Not sure I like the name of this interface, but the name isn't important. I think this is a good interface, and I like how the GZip implementation extends and implements it.

Reply:
I first had encoder. I first remember serdes used in kafka for serializer and deserializer, which is what this is doing.  I'm not sure I like either of the name choices. Open to suggestions

Reply:
In general, I think a "serialization" interface can be presumed to have a corresponding deserialization action coupled to it (a one-way serializer isn't super useful). So, I don't find it necessary to force both words into the name. I'd probably just call it a PropSerializer (or perhaps MapSerializer). Similarly, other possible names could be based on "converter" or "encoder". Or, since you named the package "codec", you could just call this "PropertyCodec", with "encode" and "decode" method names.